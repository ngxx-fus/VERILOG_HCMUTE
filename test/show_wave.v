`timescale 1ns/1ns
`include "not.v"

module show_wave;
    reg A;
    wire O;
    not_gate N(.in(A), .out(O));

    initial
        begin
            $dumpfile("show_wave.vcd");
            $dumpvars(0, show_wave);
            A = 0; 
            #20 A = 1; 
            #20 A = 0; 
            #20 A = 1; 
            #20 A = 0; 
        end


endmodule

// cd ./dir   : từ vị trí hiện tại vào bên trong thư mục dir
// cd .. : từ vị trí hiện tại, quay trở ra thư mục bố
// cls : xoa mang hinh tren terminal