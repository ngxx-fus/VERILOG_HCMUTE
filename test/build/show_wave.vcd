$date
	Wed Feb 28 17:04:10 2024
$end
$version
	Icarus Verilog
$end
$timescale
	1ns
$end
$scope module show_wave $end
$var wire 1 ! O $end
$var reg 1 " A $end
$scope module N $end
$var wire 1 " in $end
$var reg 1 ! out $end
$upscope $end
$upscope $end
$enddefinitions $end
$comment Show the parameter values. $end
$dumpall
$end
#0
$dumpvars
0"
1!
$end
#20
0!
1"
#40
1!
0"
#60
0!
1"
#80
1!
0"
