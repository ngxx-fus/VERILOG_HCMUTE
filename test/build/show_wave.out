#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 9;
:vpi_module "C:\iverilog\lib\ivl\system.vpi";
:vpi_module "C:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "C:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "C:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "C:\iverilog\lib\ivl\va_math.vpi";
S_0000019c9b855250 .scope module, "show_wave" "show_wave" 2 4;
 .timescale -9 -9;
v0000019c9b8073c0_0 .var "A", 0 0;
v0000019c9b855570_0 .net "O", 0 0, v0000019c9b806f30_0;  1 drivers
S_0000019c9b8553e0 .scope module, "N" "not_gate" 2 7, 3 1 0, S_0000019c9b855250;
 .timescale -9 -9;
    .port_info 0 /INPUT 1 "in";
    .port_info 1 /OUTPUT 1 "out";
v0000019c9b80e3d0_0 .net "in", 0 0, v0000019c9b8073c0_0;  1 drivers
v0000019c9b806f30_0 .var "out", 0 0;
E_0000019c9b80c070 .event anyedge, v0000019c9b80e3d0_0;
    .scope S_0000019c9b8553e0;
T_0 ;
    %wait E_0000019c9b80c070;
    %load/vec4 v0000019c9b80e3d0_0;
    %inv;
    %store/vec4 v0000019c9b806f30_0, 0, 1;
    %jmp T_0;
    .thread T_0, $push;
    .scope S_0000019c9b855250;
T_1 ;
    %vpi_call 2 11 "$dumpfile", "show_wave.vcd" {0 0 0};
    %vpi_call 2 12 "$dumpvars", 32'sb00000000000000000000000000000000, S_0000019c9b855250 {0 0 0};
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000019c9b8073c0_0, 0, 1;
    %delay 20, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000019c9b8073c0_0, 0, 1;
    %delay 20, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000019c9b8073c0_0, 0, 1;
    %delay 20, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000019c9b8073c0_0, 0, 1;
    %delay 20, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000019c9b8073c0_0, 0, 1;
    %end;
    .thread T_1;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "show_wave.v";
    "./not.v";
